src/appleseed/foundation/image/filteredtile.cpp

```diff
@@ -163,4 +165,57 @@ void FilteredTile::atomic_add(
     }
 }
 
+float FilteredTile::compute_weighted_pixel_variance(
+    const float*        main,
+    const float*        second)
+{
+    // Get weights.
+    float main_weight = *main++;
+    main_weight = main_weight == 0.0f ? 0.0f : 1.0f / main_weight;
```

Comment:
I suggest introducing a `rcp_main_weight` variable for clarity, and making all variables const.